src/main/java/org/example/springintro/services/impl/BookServiceImpl.java

```diff
@@ -0,0 +1,63 @@
+package org.example.springintro.services.impl;
+
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.example.springintro.dto.BookDto;
+import org.example.springintro.dto.CreateBookRequestDto;
+import org.example.springintro.exception.EntityNotFoundException;
+import org.example.springintro.mapper.BookMapper;
+import org.example.springintro.model.Book;
+import org.example.springintro.repository.BookRepository;
+import org.example.springintro.services.BookService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class BookServiceImpl implements BookService {
+    @Autowired
+    private BookRepository bookRepository;
+    private final BookMapper bookMapper;
+
+    @Override
+    public BookDto save(CreateBookRequestDto requestDto) {
+        Book book = bookMapper.toModel(requestDto);
+        return bookMapper.toDto(bookRepository.save(book));
+    }
+
+    @Override
+    public List<BookDto> findAll() {
+        return bookRepository.findAll().stream()
+                .map(bookMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public BookDto findById(Long id) {
+        Book book = bookRepository.findById(id)
+                .orElseThrow(
+                        () -> new EntityNotFoundException("Book not found by id: " + id)
+                );
+        return bookMapper.toDto(book);
+    }
+
+    @Override
+    public BookDto updateById(Long id, CreateBookRequestDto requestDto) {
+        Book book = bookRepository.findById(id)
+                .orElseThrow(
+                        () -> new EntityNotFoundException("Book not found by id: " + id)
+                );
+        book.setTitle(requestDto.getTitle());
```

Comment:
You can use Mapper method 
`bookMapper.updateBookFromDto(requestDto, book);`

where 
`void updateBookFromDto(CreateBookRequestDto book, @MappingTarget Book entity);`